cirq/circuits/circuit.py

```diff
@@ -1710,11 +1717,34 @@ def _draw_moment_in_diagram(
         if x > max_x:
             max_x = x
 
+    global_phase = np.product([
```

Comment:
this now loops twice over all ops.  Why not just keep a running product in the first loop?

Reply:
That loop is already extremely complicated, and I did not want to make it more so.

My rule of thumb is that when producing diagrams you are not in a time sensitive context, and so don't need to worry about micro-optimizing anything.

Reply:
If a loop is too complicated you should make the inner loop more readable by using other methods.